components/base/wheeled/wheeled_base.go

```diff
@@ -124,8 +158,173 @@ func (base *wheeledBase) Spin(ctx context.Context, angleDeg, degsPerSec float64,
+
+			// poll the sensor for the current error in angle
+			// check if we've overshot our target by fifteen degrees
+			// check if we've travelled at all
+			if (atTarget && minTravel) || (overShot && minTravel) {
+				ticker.Stop()
+				if err := base.Stop(base.cancelCtx, nil); err != nil {
+					return
+				}
+
+				base.logger.Debugf(
+					"stopping base with errAngle:%.2f, overshot? %t",
+					math.Abs(targetYaw-currYaw), overShot)
+				return
+			}
+		}
+	}, base.activeBackgroundWorkers.Done)
+
+	return nil
+}
+
+func getTurnState(currYaw, startYaw, targetYaw, dir, angleDeg float64) (atTarget, overShot, minTravel bool) {
+	atTarget = math.Abs(targetYaw-currYaw) < errTarget
+	overShot = hasOverShot(currYaw, startYaw, targetYaw, dir)
+	minTravel = math.Abs(currYaw-startYaw) > math.Abs(angleDeg*increment)
+	return atTarget, overShot, minTravel
+}
+
+func getCurrentYaw(ctx context.Context, ms movementsensor.MovementSensor, extra map[string]interface{},
+) (float64, error) {
+	orientation, err := ms.Orientation(ctx, extra)
+	if err != nil {
+		return 0, err
+	}
+	// Add Pi  to make the computation for overshoot simpler
+	// turns imus from -180 -> 180 to a 0 -> 360 range
+	return addAnglesInDomain(rdkutils.RadToDeg(orientation.EulerAngles().Yaw), 0), nil
+}
+
+func addAnglesInDomain(target, current float64) float64 {
+	angle := target + current
+	// reduce the angle
+	angle = math.Mod(angle, oneTurn)
+
+	// force it to be the positive remainder, so that 0 <= angle < 360
+	angle = math.Mod(angle+oneTurn, oneTurn)
+	return angle
+}
+
+func findSpinParams(angleDeg, degsPerSec, currYaw float64) (float64, float64) {
+	targetYaw := addAnglesInDomain(angleDeg, currYaw)
+	dir := 1.0
+	if math.Signbit(degsPerSec) != math.Signbit(angleDeg) {
+		// both positive or both negative -> counterclockwise spin call
+		// counterclockwise spin calls add allowable angles
+		// the signs being different --> clockwise spin call
+		// cloxkwise spin calls subtract allowable angles
+		dir = -1
+	}
+	return targetYaw, dir
+}
+
+// this function does not wrap around 360 degrees currently.
+func angleBetween(current, bound1, bound2 float64) bool {
+	if bound2 > bound1 {
+		inBewtween := current >= bound1 && current < bound2
+		return inBewtween
+	}
+	inBewteen := current > bound2 && current <= bound1
+	return inBewteen
+}
+
+func hasOverShot(angle, start, target, dir float64) bool {
+	switch {
+	case dir == -1 && start > target:
+		// for cases with a quadrant switch from 1 -> 4
+		// check if the current angle is in the regions before the
+		// target and after the start
+		over := angleBetween(angle, target, 0) || angleBetween(angle, 360, start)
+		return over
+	case dir == -1 && target > start:
+		// the overshoot range is the inside range between the start and target
+		return angleBetween(angle, target, start)
+	case dir == 1 && start > target:
+		// for cases with a quadrant switch from 1 -> 4
```

Comment:
should this say "4 -> 1"?

Reply:
either direction. adding arrows for bidirectionality.